src/plugins/visualize/public/application/components/visualize_listing.tsx

```diff
@@ -143,6 +145,12 @@ export const VisualizeListing = () => {
           }),
         });
       });
+      const uiActions = getUiActions();
+      selectedItems.forEach((item: any) => {
+        uiActions
+          .getTrigger(SAVED_OBJECT_DELETE_TRIGGER)
+          .exec({ type: 'visualization', savedObjectId: get(item, 'id', '') });
```

Comment:
A few things here, since this is a batched delete operation, this can be multiple vis objects. and they could also be partial successes.

Reply:
Try this instead.

```ts
async (selectedItems: object[]) => {
  await Promise.all(
    selectedItems.map((item: any) => savedObjects.client
      .delete(item.savedObjectType, item.id)
      .then(() => \\ send UI action per deleted object here)
  ))
})
```

Reply:
Thanks for this suggestion - updated.